src/main/java/gov/cms/madie/terminology/service/FhirTerminologyService.java

```diff
@@ -108,4 +112,81 @@ private List<QdmValueSet.Concept> getValueSetConcepts(
     log.info("No Expansion codes are found for the valueSet oid : [{}]", valueSet.getId());
     return List.of();
   }
+
+  // one to call only, one to mutate and build
+  private Bundle retrieveCodeSystemsPage(UmlsUser umlsUser, Integer offset, Integer count) {
+    IParser parser = fhirContext.newJsonParser();
+    String responseString =
+        fhirTerminologyServiceWebClient.getCodeSystemsPage(offset, count, umlsUser.getApiKey());
+      return parser.parseResource(
+            Bundle.class, responseString);
+  }
+
+  private void updateOrInsertAllCodeSystems(List<CodeSystem> codeSystemList){
+      for (CodeSystem codeSystem : codeSystemList) {
+          var id = codeSystem.getTitle() + codeSystem.getVersion();
+          Optional<CodeSystem> existingCodeSystemOptional = codeSystemRepository.findById(id);
+          if (existingCodeSystemOptional.isPresent()) {
```

Comment:
Since we're basically setting this up as a persistent cache, maybe we can just drop duplicate entries instead of modifying them. 

Off hand, I'm not sure which would be more performant.

Reply:
Modified check to             if (existingCodeSystemOptional.isEmpty()) {